tools/check_generated_backend_test_data.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+# Test backend test data
+git status
+# Skip *output_*.pb because NumPy functions might behave differently on different platforms
+# Skip test_log's input.pb because it uses np.random, which might behave differently on different platforms
+if ! git diff --exit-code -- . ':!onnx/onnx-data.proto' ':!onnx/onnx-data.proto3' ':!*output_*.pb' ':!*input_*.pb'; then
+echo "git diff for test generation returned failures. Please check updated node test files"
+exit 1
+fi
+
+if ! git diff --exit-code --diff-filter=ADR -- . ':!onnx/onnx-data.proto' ':!onnx/onnx-data.proto3'; then
+echo "Test generation returned failures. Please check the number of node test files (input_*.pb or output_*.pb)"
+exit 1
+fi
```

Comment:
Thank you for organizing `check_generated_backend_test_data.sh` and `check_generated_diff.sh`. I think we can also apply it in [MacOS-CI.yml](https://github.com/onnx/onnx/blob/main/.azure-pipelines/MacOS-CI.yml)? Windows-CI.yml is slightly different due to Windows syntax.

Reply:
Sure!

Reply:
Done